### PR TITLE
Add ModalForm in FormUI

### PR DESCRIPTION
**Pull request by S3v3Nice**

### CODE REVIEW

File: LiteLoader/Kernel/FormUI.cpp [hunk 1]
```diff
@@ -78,6 +78,52 @@ bool SimpleForm::sendTo(Player* player, Callback callback) {
...
+string ModalForm::serialize() {
+    try {
+        fifo_json form = fifo_json::parse(R"({"title":"","content":"","button1":"","button2":"","type":"modal"})");
+        form["title"] = title;
+        form["content"] = content;
+        form["button1"] = button1;
+        form["button2"] = button2;
+        return form.dump();
+    } catch (const fifo_json::exception&) {
+        logger.error("Fail to generate Simple Form in serialize!");
```
Comment (RimuruChan) [inline LiteLoader/Kernel/FormUI.cpp]: Typo: "Simple Form" should be "ModalForm"

File: LiteLoader/Header/FormUI.h [hunk 2]
```diff
@@ -4,7 +4,7 @@
 //
 //  [Example - Simple Form]
 //  - Form that contains several buttons (with optional image)
-//  - Let the player to choose one option
+//  - Lets the player to choose one option
```
Comment (RimuruChan) [inline LiteLoader/Header/FormUI.h]: Grammer: Let the player choose an option from multiple options

File: LiteLoader/Header/FormUI.h [hunk 3]
```diff
@@ -17,34 +17,49 @@
 //          [](Player* pl) { pl->sendText("Get into settings..."); })        // Settings's callback function
 //
 //      .addButton("Exit")                                                   // Add a single button "Exit"
-//      .sendTo(Level::getPlayer("John"));                   // Send the form to a player called "Jim"
+//      .sendTo(Level::getPlayer("John"));                                   // Send the form to a player called "John"
+//
+//
+//  [Example - Modal Form]
+//  - Form that contains accept button and cancell button
```
Comment (RimuruChan) [inline LiteLoader/Header/FormUI.h]: Typo: "cancell" shoud be "cancel"

Comment (RimuruChan) [reply LiteLoader/Header/FormUI.h]: Better description: Form with Cancel and Confirm buttons

File: LiteLoader/Header/FormUI.h [hunk 4]
```diff
@@ -17,34 +17,49 @@
...
+//                  player->sendText("Okay, let's go");
+//              else                                                              // Player pressed button "Nope"
+//                  player->sendText("Hm, suit yourself");
+//          });                                    
 //
 //
 //  [Example - Custom Form]
 //  - Form that contains some kinds of elements (like input line, toggle, dropdown, ....)
-//  - Let the player to provide some detailed information
+//  - Lets the player to provide some detailed information
```
Comment (RimuruChan) [inline LiteLoader/Header/FormUI.h]: Grammer: Let the player provide some detailed information

File: LiteLoader/Kernel/FormUI.cpp [hunk 5]
```diff
@@ -78,6 +78,52 @@ bool SimpleForm::sendTo(Player* player, Callback callback) {
     return player->sendRawFormPacket(id, data);
 }
 
+//////////////////////////////// Simple Form ////////////////////////////////
```
Comment (RimuruChan) [inline LiteLoader/Kernel/FormUI.cpp]: Typo: Modal Form

File: LiteLoader/Header/FormUI.h [hunk 6]
```diff
@@ -331,6 +346,33 @@ class SimpleForm : public FormImpl {
     LIAPI bool sendTo(Player* player, Callback callback = Callback());
 };
 
+class ModalForm : public FormImpl {
+protected:
+    LIAPI string serialize() override;
+
+public:
+    using Callback = std::function<void(Player*, bool)>;
+    string title, content, button1, button2;
```
Comment (RimuruChan) [inline LiteLoader/Header/FormUI.h]: Change to more specific and canonical naming

**Review by RimuruChan — CHANGES_REQUESTED** [review]

File: LiteLoader/Header/FormUI.h (hunk 6 above)
Comment (S3v3Nice) [reply LiteLoader/Header/FormUI.h]: button1Text and button2Text? Or confirmButton and cancelButton?

**Review by ShrBox — APPROVED** [review]

**Review by RimuruChan — APPROVED** [review]

**Pull request merged**